src/kvstore/raftex/RaftPart.cpp

```diff
@@ -954,36 +1001,73 @@ void RaftPart::processAppendLogResponses(
         this->appendLogsInternal(std::move(iter), currTerm);
     } else {
         // Not enough hosts accepted the log, re-try
-        LOG_EVERY_N(WARNING, 100) << idStr_ << "Only " << numSucceeded
-                                  << " hosts succeeded, Need to try again";
-        replicateLogs(eb,
-                      std::move(iter),
-                      currTerm,
-                      lastLogId,
-                      committedId,
-                      prevLogTerm,
-                      prevLogId);
+        retryNum_++;
+        std::stringstream ss;
+        for (auto& res : resps) {
+            ss << "[" << hosts[res.first] << "] err code is "
+               << static_cast<int32_t>(res.second.get_error_code())
+               << ", ";
+        }
+        LOG(WARNING) << idStr_ << "Only " << numSucceeded
+                               << " hosts succeeded, Need to try again after "
+                               << retryNum_ << "ms"
+                               << ", errMsg " << ss.str();
+
+        auto task = [self = shared_from_this(),
+                     iter = std::move(iter),
+                     eb,
+                     currTerm,
+                     lastLogId,
+                     committedId,
+                     prevLogTerm,
+                     prevLogId] () mutable {
+            self->replicateLogs(eb,
+                                std::move(iter),
+                                currTerm,
+                                lastLogId,
+                                committedId,
+                                prevLogTerm,
+                                prevLogId);
+        };
+
+        folly::via(eb, [eb,
+                        retryNum = retryNum_,
+                        task = std::move(task)] () mutable {
+            eb->runAfterDelay(std::move(task), retryNum);
```

Comment:
is the runAfterDelay is function:
/**
   * Runs the given Cob at some time after the specified number of
   * milliseconds.  (No guarantees exactly when.)
   *
   * Throws a std::system_error if an error occurs.
   */
  void runAfterDelay(
      Func cob,
      uint32_t milliseconds,
      InternalEnum internal = InternalEnum::NORMAL);